regcomp.c

```diff
@@ -12622,54 +12651,31 @@ S_regpiece(pTHX_ RExC_state_t *pRExC_state, I32 *flagp, U32 depth)
 
       case '{':  /* A '{' may or may not indicate a quantifier; call regcurly()
                     to determine which */
-        if (regcurly(RExC_parse)) {
-            const char* endptr;
-
-            /* Here is a quantifier, parse for min and max values */
-            maxpos = NULL;
-            next = RExC_parse + 1;
-            while (isDIGIT(*next) || *next == ',') {
-                if (*next == ',') {
-                    if (maxpos)
-                        break;
-                    else
-                        maxpos = next;
-                }
-                next++;
-            }
+        if (regcurly(RExC_parse, RExC_end, regcurly_return)) {
+            const char * min_start = regcurly_return[1];
+            const char * min_end   = regcurly_return[2];
+            const char * max_start = regcurly_return[3];
+            const char * max_end   = regcurly_return[4];
 
-            assert(*next == '}');
+            assert(min_start);
+            min = get_quantifier_value(pRExC_state, min_start, min_end);
```

Comment:
Was this supposed to handle `if (min_start == min_end) min = 0; else ...`? It doesn't look like `get_quantifier_value` handles that, but the `{,n}` comment below implies that something should.

Reply:
The comment is wrong, and should be changed.  Here there should be an assert about min_end.